### PR TITLE
Worker register SDK changes, tests and optimization

### PR DESCRIPTION
**Pull request by shresthichauhan**
Test Name | Listener | SDK
-- | -- | --
test_workerregister_workerId_null | TCFH-810 | TCFH-810
test_workerregister_hashingAlgorithm_KECCAK-256 | Pass | Pass
test_workerregister_hashingAlgorithm_alternate | TCFH-810 | TCFH-810
test_workerregister_signingAlgorithm_RSA-OAEP-3072 | Pass | Pass
test_workerregister_signingAlgorithm_alternate | TCFH-810 | TCFH-810
test_workerregister_keyEncryptionAlgorithm_alternate | TCFH-810 | TCFH-810
test_workerregister_dataEncryptionAlgorithm_list | TCFH-810 | TCFH-810
test_workerregister_dataEncryptionAlgorithm_alternate | TCFH-810 | TCFH-810
test_workerregister_workerType_invalid | TCFH-810 | NA
test_workerregister_organizationId_empty | TCFH-810 | TCFH-810
test_workerregister_applicationTypeId_empty | TCFH-810 | TCFH-810
test_workerregister_orgnizationid_32bytes | Pass | Pass
test_workerregister_applicationTypeId_32bytes | Pass | Pass
test_workerregister_workOrderPayloadFormats_JSON-RPC-JWT | TCFH-810 | TCFH-810

**Note**: Require to change (/usr/local/lib/python3.6/dist-packages/avalon_sdk/connector/direct/jrpc/jrpc_worker_registry.py) the details field to get the actual test status for SDK tests. Otherwise 2 worker register valid tests will fail. It will start passing when fix will be in production code. Refer TCFH-883.

### CODE REVIEW

File: tests/automation_framework/data/worker/workerregister_applicationTypeId_32bytes.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+    "jsonrpc": "2.0",
+    "method": "WorkerRegister",
+    "id": 1,
+    "params": {
+        "workerId":"",
+        "workerType": 1,
+        "organizationId":"0xAD",
+        "applicationTypeId": "",
```
Comment (jinengandhi-intel) [inline tests/automation_framework/data/worker/workerregister_applicationTypeId_32bytes.json:9]: Why is the applicationTypeId set to empty here when the test case is for 32 bytes?

File: tests/automation_framework/data/worker/workerregister_organizationId_empty.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+    "jsonrpc": "2.0",
+    "method": "WorkerRegister",
+    "id": 1,
+    "params": {
+        "workerId":"",
+        "workerType": 1,
+        "organizationId":" ",
```
Comment (jinengandhi-intel) [inline tests/automation_framework/data/worker/workerregister_organizationId_empty.json:8]: Here "organizationId":" " is not empty but I see a space between opening and closing double quotes.

File: tests/automation_framework/data/worker/workerregister_orgnizationid_32bytes.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+    "jsonrpc": "2.0",
+    "method": "WorkerRegister",
+    "id": 1,
+    "params": {
+        "workerId":"",
+        "workerType": 1,
+        "organizationId":"",
```
Comment (jinengandhi-intel) [inline tests/automation_framework/data/worker/workerregister_orgnizationid_32bytes.json:8]: Set it to the right 32 bytes value, it shouldn't be empty.

File: tests/automation_framework/data/worker/workerregister_applicationTypeId_empty.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{
+    "jsonrpc": "2.0",
+    "method": "WorkerRegister",
+    "id": 1,
+    "params": {
+        "workerId":"",
+        "workerType": 1,
+        "organizationId":"",
+        "applicationTypeId": " ",
```
Comment (jinengandhi-intel) [inline tests/automation_framework/data/worker/workerregister_applicationTypeId_empty.json:9]: Here "applicationTypeId":" " is not empty but I see a space between opening and closing double quotes.

File: tests/automation_framework/data/worker/workerregister_signingAlgorithm_RSA_OAEP_3072.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (jinengandhi-intel) [inline tests/automation_framework/data/worker/workerregister_signingAlgorithm_RSA_OAEP_3072.json:1]: Rename the file to workerregister_signingAlgorithm_RSAOAEP3072.json, the last part should be one word as per the naming conventions.

File: tests/automation_framework/data/worker/workerregister_workOrderPayloadFormats_JSON_RPC_JWT.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (jinengandhi-intel) [inline tests/automation_framework/data/worker/workerregister_workOrderPayloadFormats_JSON_RPC_JWT.json:1]: Rename the file to workerregister_workOrderPayloadFormats_JSONRPCJWT.json as per the naming conventions.

File: tests/automation_framework/data/worker/workerregister_workerEncryptionKey_empty.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
+    "jsonrpc": "2.0",
+    "method": "WorkerRegister",
+    "id": 1,
+    "params": {
+        "workerId":"",
+        "workerType": 1,
+        "organizationId":"",
+        "applicationTypeId": "0xAB",
+        "workerEncryptionKey" : " " ,
```
Comment (jinengandhi-intel) [inline tests/automation_framework/data/worker/workerregister_workerEncryptionKey_empty.json]: Again like before workerEncryptionKey is not really empty.

**Review by jinengandhi-intel — CHANGES_REQUESTED** [review]

File: tests/automation_framework/data/worker/workerregister_workOrderPayloadFormats_JSON_RPC_JWT.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        "workerType": 1,
+        "organizationId":"0xAD",
+        "applicationTypeId": "0XAD",
+        "workerEncryptionKey" : " " ,
+        "details": {
+            "hashingAlgorithm": "SHA-256",
+            "signingAlgorithm": "SECP256K1" ,
+            "keyEncryptionAlgorithm": "RSA-OAEP-3072",
+            "dataEncryptionAlgorithm": "AES-GCM-256",
+	    "workOrderPayloadFormats": ["JSON-RPC-JWT"],
```
Comment (manojsalunke85) [inline tests/automation_framework/data/worker/workerregister_workOrderPayloadFormats_JSON_RPC_JWT.json:15]: "workOrderPayloadFormats" is not aligned in line with the other parameters above.

Comment (manojsalunke85) [reply tests/automation_framework/data/worker/workerregister_workOrderPayloadFormats_JSON_RPC_JWT.json:15]: Do you also want to add a test for workOrderPayloadFormats: ["JSON-RPC"]

**Review by jinengandhi-intel — APPROVED** [review]

**Pull request merged**